framework/src/org/checkerframework/common/reflection/ClassValAnnotatedTypeFactory.java

```diff
@@ -244,18 +236,16 @@ public Void visitMethodInvocation(MethodInvocationTree tree,
         }
 
         private boolean isForNameMethodInovaction(MethodInvocationTree tree) {
-            for (ExecutableElement method: forName ) {
-                if (TreeUtils.isMethodInvocation(tree, method, processingEnv)) {
-                    return true;
-                }
+            if (getDeclAnnotation(InternalUtils.symbol(tree),
+                    ForName.class) != null) {
+                return true;
             }
            return false;
         }
         private boolean isGetClassMethodInovaction(MethodInvocationTree tree) {
-            for (ExecutableElement method: getClass ) {
-                if (TreeUtils.isMethodInvocation(tree, method, processingEnv)) {
-                    return true;
-                }
+            if (getDeclAnnotation(InternalUtils.symbol(tree),
+                    GetClass.class) != null) {
+                return true;
```

Comment:
could be re-written as:

return getDeclAnnotation(InternalUtils.symbol(tree), GetClass.class) != null;